tests/unit_tests/test_swap_by_newest.py

```diff
@@ -0,0 +1,105 @@
+import pytest
```

Comment:
One thing that I love about this particular set of test suites is that y'all recognized that it is going to be pretty similar to the suite of tests that test the category comparison functions. Y'all decided to follow the same structure and test similar ideas which mimics what you will definitely see in the industry! Don't reinvent the wheel! Follow the patterns you see elsewhere and you'll be good to go!